narwhal/worker/src/batch_maker.rs

```diff
@@ -126,7 +167,13 @@ impl BatchMaker {
                         ReconfigureNotification::Shutdown => return
                     }
                     tracing::debug!("Committee updated to {}", self.committee);
-                }
+                },
+
+                // Process the pipeline of batches, this consumes items in the `batch_pipeline`
+                // list, and ensures the main loop in run will always be able to make progress
+                // by lowering it until condition batch_pipeline.len() < MAX_PARALLEL_BATCH is met.
+                _ = batch_pipeline.next(), if !batch_pipeline.is_empty() => {}
```

Comment:
Should we decrement `parallel_worker_batches` here?

Reply:
I now update the gauge in the body `self.node_metrics.parallel_worker_batches.set(batch_pipeline.len() as i64);`